tests/unit/ApprovalType.spec.ts

```diff
@@ -0,0 +1,133 @@
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+import { createLocalVue, mount, Wrapper } from '@vue/test-utils'
+import { ApprovalType } from '@/components/approval-type'
+import VueRouter from 'vue-router'
+
+Vue.use(Vuetify)
+let vuetify = new Vuetify({})
+
+const localVue = createLocalVue()
+localVue.use(VueRouter)
+
+/**
+ * Creates and mounts a blank, un-populated component
+ */
+function createDefaultComponent (
+  draftCourtOrderNumber: string = '',
+  approvedByRegistrar: boolean = false,
+  isConversionToFullRestoration: boolean = false
+): Wrapper<ApprovalType> {
+  return mount(ApprovalType, {
+    propsData: {
+      draftCourtOrderNumber,
+      approvedByRegistrar,
+      isConversionToFullRestoration
+    },
+    vuetify,
+    localVue
+  })
+}
+
+describe('Initialize ApprovalType component', () => {
+  it('loads the component', async () => {
+    const wrapper: Wrapper<ApprovalType> = createDefaultComponent()
+    expect(wrapper.findComponent(ApprovalType).exists()).toBe(true)
+    wrapper.destroy()
+  })
+
+  it('component default state has no fields populated', async () => {
+    const wrapper: Wrapper<ApprovalType> = createDefaultComponent()
+    expect(wrapper.vm.$data.courtOrderNumber).toBe('')
+    expect(wrapper.vm.$data.approvalTypeSelected).toBe('')
+    expect(wrapper.vm.$data.valid).toBe(true)
+    wrapper.destroy()
+  })
+
+  it('if isConversionToFullRestoration prop true, alternative wording shown', async () => {
+    const wrapper: Wrapper<ApprovalType> = createDefaultComponent()
+    expect(wrapper.find('label.v-label.theme--light').text())
+      .toEqual('This restoration is approved by court order.')
+
+    // Prompt validates through prop
+    wrapper.setProps({
+      isConversionToFullRestoration: true
+    })
+    await Vue.nextTick()
+    expect(wrapper.find('label.v-label.theme--light').text())
+      .toEqual('This conversion to full restoration is approved by court order.')
+    wrapper.destroy()
+  })
+
+  it('component emits events when court order number entered', async () => {
+    const wrapper: Wrapper<ApprovalType> = createDefaultComponent()
+    // Input text into text-field
+    const input = wrapper.find('#court-order-number-input')
+    input.setValue('89123456')
+    await Vue.nextTick()
```

Comment:
I'm not certain if it will work here, but sometimes you can simply await the setValue() above.

Sometimes you need to wait for the next timeslice (remember, JS is single-threaded) so that validation can take place and set the properties you expect to see changed.

Reply:
Good to know.  If it's okay for you, I'll leave this as is.